### PR TITLE
[docs] Clarify nature of match for `match` expression in Style Spec

### PR DESCRIPTION
**Pull request by domlet**
Closes https://github.com/mapbox/mapbox-gl-js-docs/issues/174

* In [Style Spec docs](https://docs.mapbox.com/mapbox-gl-js/style-spec/expressions/#match), clarifies match type with:
  > The input matches if any of the values in the array matches using `===`

* Copyedits to the same description.

Before/After Screenshot:

Screen Shot 2021-03-18 at 8 00 33 AM

 - [x] briefly describe the changes in this PR
 - [x] include before/after visuals or gifs if this PR includes visual changes
 - [x] tagged @mapbox/map-design-team `@mapbox/static-apis` if this PR includes style spec API or visual changes
 - [ ] tagged `@mapbox/gl-native` if this PR includes shader changes or needs a native port
 - [x] apply changelog label ('bug', 'feature', 'docs', etc) or use the label 'skip changelog'
- [ ] Reviews by @HeyStenson @ansis

### CODE REVIEW

File: src/style-spec/reference/v8.json [hunk 1]
```diff
@@ -2733,7 +2733,7 @@
...
-        "doc": "Selects the output whose label value matches the input value, or the fallback value if no match is found. The input can be any expression (e.g. `[\"get\", \"building_type\"]`). Each label must be either:\n - a single literal value; or\n - an array of literal values, whose values must be all strings or all numbers (e.g. `[100, 101]` or `[\"c\", \"b\"]`). The input matches if any of the values in the array matches, similar to the `\"in\"` operator.\nEach label must be unique. If the input type does not match the type of the labels, the result will be the fallback value.",
+        "doc": "Selects the output whose label value matches the input value, or the fallback value if no match is found. The input can be any expression (e.g. `[\"get\", \"building_type\"]`). Each label must be unique, and must be either:\n - a single literal value; or\n - an array of literal values, whose values must be all strings or all numbers (e.g. `[100, 101]` or `[\"c\", \"b\"]`).\n\nThe input matches if any of the values in the array matches using `===`, similar to the `\"in\"` operator.\nIf the input type does not match the type of the labels, the result will be the fallback value.",
```
Comment (asheemmamoowala) [inline src/style-spec/reference/v8.json:2736]: While the `===` reference makes sense for Javascript developers, the Style spec is programming language independent and is used in our Mobile SDKs as well. Would `strict equality` be an adequate replacement?

Comment (domlet) [inline src/style-spec/reference/v8.json:2736]:
```suggestion
        "doc": "Selects the output whose label value matches the input value, or the fallback value if no match is found. The input can be any expression (e.g. `[\"get\", \"building_type\"]`). Each label must be unique, and must be either:\n - a single literal value; or\n - an array of literal values, whose values must be all strings or all numbers (e.g. `[100, 101]` or `[\"c\", \"b\"]`).\n\nThe input matches if any of the values in the array matches using strict equality, similar to the `\"in\"` operator.\nIf the input type does not match the type of the labels, the result will be the fallback value.",
```

Comment (domlet) [reply src/style-spec/reference/v8.json]: Absolutely: 6a7c755 Thank you @asheemmamoowala for making this more clear!

File: src/style-spec/reference/v8.json [hunk 2]
```diff
@@ -2733,7 +2733,7 @@
...
+        "doc": "Selects the output whose label value matches the input value, or the fallback value if no match is found. The input can be any expression (e.g. `[\"get\", \"building_type\"]`). Each label must be unique, and must be either:\n - a single literal value; or\n - an array of literal values, whose values must be all strings or all numbers (e.g. `[100, 101]` or `[\"c\", \"b\"]`).\n\nThe input matches if any of the values in the array matches using strict equality, similar to the `\"in\"` operator.\nIf the input type does not match the type of the labels, the result will be the fallback value.",
```
Comment (HeyStenson) [inline src/style-spec/reference/v8.json:2736]:
```suggestion
        "doc": "Selects the output for which the label value matches the input value, or the fallback value if no match is found. The input can be any expression (for example, `[\"get\", \"building_type\"]`). Each label must be unique, and must be either:\n - a single literal value; or\n - an array of literal values, the values of which must be all strings or all numbers (for example `[100, 101]` or `[\"c\", \"b\"]`).\n\nThe input matches if any of the values in the array matches using strict equality, similar to the `\"in\"` operator.\nIf the input type does not match the type of the labels, the result will be the fallback value.",
```

Comment (HeyStenson) [inline src/style-spec/reference/v8.json:2736]:
```suggestion
        "doc": "Selects the output for which the label value matches the input value, or the fallback value if no match is found. The input can be any expression (for example, `[\"get\", \"building_type\"]`). Each label must be unique, and must be either:\n - a single literal value; or\n - an array of literal values, the values of which must be all strings or all numbers (for example `[100, 101]` or `[\"c\", \"b\"]`).\n\nThe input matches if any of the values in the array matches using strict equality, similar to the `\"in\"` operator.\nIf the input type does not match the type of the labels, the result will be the fallback value.",
```

**Review by HeyStenson — APPROVED** [review]
Looks good @domlet!

File: src/style-spec/reference/v8.json (hunk 2 above)
Comment (domlet) [reply src/style-spec/reference/v8.json]: (Resolving this conversation because this set of suggestions is a duplicate; already committed in 3676089.

**Review by asheemmamoowala — APPROVED** [review]

**Pull request merged**